now/app/text_to_text_and_image/app.py

```diff
@@ -149,9 +167,11 @@ def setup(
             kubectl_path=kubectl_path,
             deployment_type=user_input.deployment_type,
         )
-        env_dict['HOSTS'] = indexer_config['hosts']
+        env_dict['HOSTS'] = indexer_config.get('hosts')
         env_dict['INDEXER_NAME'] = f"jinahub+docker://{indexer_config['indexer_uses']}"
-        env_dict['INDEXER_MEM'] = indexer_config['indexer_resources']['INDEXER_MEM']
+        env_dict['INDEXER_MEM'] = indexer_config.get('indexer_resources').get(
+            'INDEXER_MEM'
+        )
```

Comment:
if you want to avoid `KeyError` and return `None` if there's no indexer resources:
```suggestion
        env_dict['INDEXER_MEM'] = indexer_config.get('indexer_resources', {}).get(
            'INDEXER_MEM'
        )
```

Reply:
Done.